src/components/Element/Element.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Basic Element that can be used as underlying component for composition.
+ */
+const Element = ({ as, deepRef, ...props }) =>
```

Comment:
Haha. Now you're doing the same thing as the `HtmlElement` component. 😆 I was somewhat hoping you could localize this component to the Tabs/Tab, where you need it so we don't create global components like this.

Additionally, I would suggest you limit the `as` prop to some specific strings or a React element. Then you can map the string values to React elements you have already rendered out in a map. The elements you are just going to render out.

Somehow I remember having these createElement calls would not be so great for performance, but I might have it wrong. Maybe this was also related more to styled components? Dunno.

Reply:
I don't mind moving it locally under the tab tree, at all :)

The `createElement` is specifically for the `Link` component in the dashboard. There I think we need the whole tab to be working anchor and to work as a navigation. This is the reason for the whole thing.

I don't think there is performance change, AFAIK. Any `<Component {...props. />` is React.createElement(Component, props) under the hood anyway.